labs/authentication-passport/README.md

```diff
@@ -0,0 +1,185 @@
+# Passport Strategy Adapter
+
+_Important: We suggest users understand LoopBack's authentication system[Link
+TBD](some loopback.io link) before using this module_
+
+This is an adapter module created for plugging in
+[`passport`](https://www.npmjs.com/package/passport) base strategies to the
+authentication system in `@loopback/authentication@2.x`.
+
+## Installation
+
+```sh
+npm i @loopback/passport-adapter --save
+```
+
+## Background
+
+`@loopback/authentication@2.x` allows users to register authentication
+strategies that implement the interface
+[`AuthenticationStrategy`](https://apidocs.strongloop.com/@loopback%2fdocs/authentication.html#AuthenticationStrategy)
+
+Since `AuthenticationStrategy` describes a strategy with different contracts
+than the passport
+[`Strategy`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/passport/index.d.ts#L79),
+and we'd like to support the existing 500+ community passport strategies, an
+**adapter class** is created in this package to convert a passport strategy to
+the one that LoopBack 4 authentication system wants.
+
+## Usage
+
+### Simply Usage
+
+1. Created an instance of the passport strategy
+
+Take the basic strategy exported from
+[`passport-http`](https://github.com/jaredhanson/passport-http) as an example,
+first create an instance of the basic strategy with your `verify` function.
+
+```ts
+function verify(username: string, password: string, cb: Function) {
+  users.find(username, password, cb);
+}
+const basicStrategy = new BasicStrategy(verify);
+```
+
+It's the similar configuration as you do when adding a strategy to a `passport`
+by calling `passport.use()`.
+
+But don't provide any passport specific options since we don't support the
+session manager.
+
+2. Apply the adapter to the strategy
+
+```ts
+const AUTH_STRATEGY_NAME = 'basic';
+
+const basicAuthStrategy = new StrategyAdapter(
+  // The configured basic strategy instance
+  basicStrategy,
+  // Give the strategy a name
+  // You'd better define your strategy name as a constant, like
+  // `const AUTH_STRATEGY_NAME = 'basic'`.
+  // So that you will decorate the APIs later with the same name.
+  AUTH_STRATEGY_NAME,
+);
+```
+
+3. Register(bind) the strategy to app
+
+```ts
+import {Application, CoreTags} from '@loopback/core';
+import {AuthenticationBindings} from '@loopback/authentication';
+
+app
+  .bind('authentication.strategies.basicAuthStrategy')
```

Comment:
> .bind('authentication.strategies.basicAuthStrategy') 

Is there a constant that can be used from @authenticate to form the first two segments of this string?

Reply:
e.g. https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/keys.ts#L108

Reply:
I guess the class name of the strategy here 

https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/types.ts#L65 

is appended as the last segment of the string?

Reply:
@emonddr The key itself could be any name, it is the **tag** that makes the strategy recognized by the extension point, and the **`name`** property defined in the strategy class that needs to match the name in decorator `@authenticate('basic')`

I don't have a strong opinion on using a string or leverage the constant in  https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/keys.ts#L108, so I would respect your preference.